great_expectations/datasource/fluent/sources.py

```diff
@@ -386,7 +387,12 @@ def update_datasource(name: str, **kwargs) -> Datasource:
             return self.create_add_crud_method(datasource_type)(name=name, **kwargs)
 
         update_datasource.__doc__ = doc_string
-        # TODO: update signature
+        update_datasource.__signature__ = _merge_signatures(
+            update_datasource,
+            datasource_type,
```

Comment:
@Kilo59 Quick question -- does `datasource_type` have a `__call__()` method -- i.e., is it callable?  Thanks.

Reply:
Where are merging the `datasource_type.__init__` method.